### PR TITLE
feat: flake it till you make it

### PR DESCRIPTION
**Pull request by hugosenari**

### CODE REVIEW

**Comment by hugosenari** [comment]
@Melkor333 can you help me test this, for some reason it freeze my machine at boot :(

File: README.md [hunk 1]
```diff
@@ -5,39 +5,66 @@ This repo contains a plymouth theme for Nixos, thanks to [discourse](https://dis
...
+## Flakes
+
+You can include it in your `flakes.nix` like this:
+
+```nix
+{
+  inputs.nixos-boot.url = "github:Melkor333/nixos-boot";
+  outputs = { self, nixpkgs, nixos-boot }:
+  {
+    nixosConfigurations."<hostname>" = inputs.nixpkgs.lib.nixosSystem {
```
Comment (hugosenari) [inline README.md]:
```suggestion
    nixosConfigurations."<hostname>" = nixpkgs.lib.nixosSystem {
```

**Comment by Melkor333** [comment]
woww! That looks really nice :-O

Sorry for my silence‚ got a lot going on… I took a brief look yesterday but I'll need to look at it with a fresh mind and a bit of time (which can take a few more days :-/ ).

FWIW the module looks good enough that I'd say we try to also add this to nixpkgs when it works!

File: modules.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{ pkgs, lib, config, ...}:
+let
+  toBG = { red, green, blue }:
+    "${toString(255.0 / red)}, "   +
+    "${toString(255.0 / green)}, " +
+    "${toString(255.0 / blue)})";
```
Comment (hugosenari) [inline modules.nix]: This logic isn't correct.

Should be:

```suggestion
    "${toString(red   / 255.0)}, "   +
    "${toString(green / 255.0)}, " +
    "${toString(blue  / 255.0)})";
```

```bash
nix-repl> 255 / 255.0
1

nix-repl> 200 / 255.0
0.784314

nix-repl> 150 / 255.0
0.588235

nix-repl> 100 / 255.0
0.392157

nix-repl> 1 / 255.0
0.00392157

nix-repl> 0 / 255.0
0  # I wasnt expecting this, but is ok for our case
```

**Comment by Melkor333** [comment]
Does your system use EFI? Maybe that's the reason for the vm to freeze. It might make sense to write a second config with grub but Idk how to force efi/bios for the vm (yet).

I don't get an actual working plymouth screen but the vm starts up fine. I can't login to the testvm as a test user though, something is fishy!

I will take another look next week, if the vm works it's possible to run `plymouth` interactively after logging in. IIRC that will also show some error messages if something's wrong with e.g. the plymouth script. (or just look at `journalctl -eu plymouthd`)

File: modules.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{ pkgs, lib, config, ...}:
+let
+  toBG = { red, green, blue }:
+    "${toString(red   / 255.0)}, "   +
+    "${toString(green / 255.0)}, " +
+    "${toString(blue  / 255.0)})";
```
Comment (Melkor333) [inline modules.nix]: this should be `"${toString(blue  / 255.0)}";` (no `)` at the end).

FWIW when we're already doing some magic I'd rather have a `backgroundcolor` value which is hex like `#FFFFFF` and then convert that. Something like [this](https://github.com/bertof/nix-rice/blob/main/color.nix#L40C1-L40C94) might work. 🤔 
If you don't feel like tinkering and just remove the `)`, that's also fine though

Comment (hugosenari) [reply modules.nix]: Liked the idea of using nix-rice

I will fix it for now and add rice the next PR
Thanks for that

Comment (hugosenari) [reply modules.nix]:
```suggestion
    "${toString(blue  / 255.0)}";
```

Comment (hugosenari) [reply modules.nix]: Options:
- Copy and paste from nix-rice
- Import it with flake (will work only in flakes)
- Import it without flake (set hash for nix-rice in the module)
- Other?

Comment (Melkor333) [reply modules.nix]: Given we only need a single line (AFAIU) and it's MIT as well I'd just copypaste and prepend with a comment `# thanks to https://github.com/bertof/nix-rice`. :)

Comment (hugosenari) [reply modules.nix]: @Melkor333  see  #10

I did not even tested it, only to joke about your comment :-D

**Pull request merged**